internal/fs/dir_handle.go

```diff
@@ -156,78 +173,64 @@ func fixConflictingNames(entries []fuseutil.Dirent) (err error) {
 	return
 }
 
-// Read all entries for the directory, fix up conflicting names, and fill in
-// offset fields.
-//
-// LOCKS_REQUIRED(in)
-func readAllEntries(
-	ctx context.Context,
-	in inode.DirInode) (entries []fuseutil.Dirent, err error) {
-	// Read one batch
-	var batch []fuseutil.Dirent
-	batch, ContinuationToken, err = in.ReadEntries(ctx, ContinuationToken)
-	if err != nil {
-		err = fmt.Errorf("ReadEntries: %w", err)
-		return
-	}
-	// Accumulate.
-	entries = append(entries, batch...)
-
-	// Ensure that the entries are sorted, for use in fixConflictingNames
-	// below.
-	// TODO: Fix this after  asynchronous fetch is added.
-	sort.Sort(sortedDirents(entries))
-
-	// Fix name conflicts.
-	err = fixConflictingNames(entries)
-	if err != nil {
-		err = fmt.Errorf("fixConflictingNames: %w", err)
-		return
-	}
-
-	// Return a bogus inode ID for each entry, but not the root inode ID.
-	//
-	// NOTE(jacobsa): As far as I can tell this is harmless. Minting and
-	// returning a real inode ID is difficult because fuse does not count
-	// readdir as an operation that increases the inode ID's lookup count and
-	// we therefore don't get a forget for it later, but we would like to not
-	// have to remember every inode ID that we've ever minted for readdir.
-	//
-	// If it turns out this is not harmless, we'll need to switch to something
-	// like inode IDs based on (object name, generation) hashes. But then what
-	// about the birthday problem? And more importantly, what about our
-	// semantic of not minting a new inode ID when the generation changes due
-	// to a local action?
-	for i, _ := range entries {
-		entries[i].Inode = fuseops.RootInodeID + 1
-	}
-
-	return
-}
+func (dh *DirHandle) FetchEntriesAsync(
+	rootInodeId int,
+	firstCall bool) {
```

Comment:
Why do you need firstCall variable? Can't you figure it our from offset? e.g. offset=0

Reply:
Because the Offset is updated in the goroutine serving kernel request, there is a possibility that this goroutine ends up in a continuous loop even when the responses returned is empty.This is an easy way to identify the first call as well as useful for ease of testing.

Reply:
can we use a do-while type loop here to avoid firstCall parameter?

Reply:
Done.